aspnetcore/security/authorization/iauthorizationpolicyprovider.md

```diff
@@ -155,11 +156,11 @@ public Task<AuthorizationPolicy> GetDefaultPolicyAsync() =>
 
 As with all aspects of a custom `IAuthorizationPolicyProvider`, you can customize this, as needed. In some cases, it may be desirable to retrieve the default policy from a fallback `IAuthorizationPolicyProvider`.
 
-## Required policy
+## Fallback policy
 
-A custom `IAuthorizationPolicyProvider` needs to implement `GetRequiredPolicyAsync` to, optionally, provide a policy that is always required. If `GetRequiredPolicyAsync` returns a non-null policy, that policy will be combined with any other (named or default) policy that is requested.
+A custom `IAuthorizationPolicyProvider` must optionally implement `GetFallbackPolicyAsync` to provide a policy that's used by the Authorization Middleware when no policies are specified. If `GetFallbackPolicyAsync` returns a non-null policy, that policy is used by the Authorization Middleware when no policies are specified for the request.
```

Comment:
Must optionally? It's either optionally or must, it can't be both.

Reply:
I just fixed that.

Reply:
I'd expand this, compare it to say "[Authorize(Policy="MyRequiredPolicy")]" for clarity?

Reply:
The confusion is probably because you can implement it to return null always, hence you MUST implement the API but you can optionally have it do nothing (basically disable the feature)

Reply:
The comparison is subtle, but sure I guess we can say that having a fallbackpolicy return "MyRequiredPolicy" would be equivalent to adding `[Authorize("MyRequiredPolicy")]` by default?